learning.py

```diff
@@ -413,7 +409,7 @@ def decision_list_learning(examples):
             return [(True, False)]
         t, o, examples_t = find_examples(examples)
         if not t:
-            raise Failure
+            raise Exception
```

Comment:
This should be something more explicit that Exception. Maybe ValueError if there is nothing better.